src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -47,5 +47,41 @@ private static AbstractManagedIdentity SelectManagedIdentitySource(RequestContex
                 AzureArcManagedIdentitySource.TryCreate(requestContext) ??
                 new ImdsManagedIdentitySource(requestContext);
         }
+
+        internal static ManagedIdentitySource GetManagedIdentitySource()
+        {
+            string identityEndpoint = EnvironmentVariables.IdentityEndpoint;
```

Comment:
this logic already exists in MSAL in those "tryCreate" methods, can it be reused?

CC @Robbie-Microsoft

Reply:
Refactored a bit to separate the logic for detect and validate.